src/DotNetLightning.Core/Utils/SeqConsumer.fs

```diff
@@ -0,0 +1,83 @@
+namespace DotNetLightning.Utils
+
+open ResultUtils
+open ResultUtils.Portability
+
+[<AutoOpen>]
+module SeqConsumerCE =
+    type SeqConsumer<'SeqElement, 'T> = {
```

Comment:
An idea of `SeqConsumer` is quite neat and I like it. sounds like simple version of parser combinator.
Would you mind adding simple explanation document to the type?

Reply:
That's what it is, yeah. Though I didn't think of that while I was writing it. Maybe I should rename it `SeqParser`?

Reply:
can we also rename `'T` to something longer please?

Reply:
> Maybe I should rename it SeqParser?

Yes, I think `SeqParser` is more easy to understand. consuming does not imply parsing.

Reply:
Fixed. I've renamed it `SeqParser`. Also the generic argument is now called `'Value` instead `'T`.